include/ensmallen_bits/nsga2/nsga2_impl.hpp

```diff
@@ -365,6 +365,8 @@ inline void NSGA2::FastNonDominatedSort(
     i++;
     fronts.push_back(nextFront);
   }
+  //Remove the empty final set
+  fronts.pop_back();
```

Comment:
Caught a big fish here. Allow me to explain myself,

- The while loop breaks when the newFront is empty. Note that the newFront is pushed **regardless** of any of the conditions.
- At the end of our while loop, the nextFront is not updated at all! That means its an empty vector which gets pushed into our ```front```
-  This causes the while loop to terminate

Result?

You got an additional empty vector in your ```front``` set. Which causes smelly code.

Reply:
https://github.com/haris989/NSGA-II/blob/7680fd70c5202d0673aa26b9fb50a8e9c387623b/NSGA%20II.py#L71

Reply:
Nice catch!